Bottom: 0bc87049cf88620c64493a3956e4f158b647ed6c
Top:    356c714d392f1f494549e247dfc7f59cbdd3a99b
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-06 17:27:02 +0000

draw_funcs.c: Don't draw pins/vias in layer for non GUI exporters

The non-gui paths render pins and vias with draw_ppv, so no need to
also render them for each layer.


---

diff --git a/src/draw_funcs.c b/src/draw_funcs.c
index 30e42ac..bd3df87 100644
--- a/src/draw_funcs.c
+++ b/src/draw_funcs.c
@@ -430,6 +430,10 @@ draw_layer (LayerType *layer, const BoxType *drawn_area, void *userdata)
   if (layer_num >= max_copper_layer)
     return;
 
+  /* Pins, pads and vias are handled in elsewhere by exporters, don't duplicate */
+  if (!gui->gui)
+    return;
+
   r_search (PCB->Data->pin_tree, drawn_area, NULL, pin_inlayer_callback, layer);
 
   /* draw element pads */
